IGNITE-17141 Use maven for build .NET and update versions (#10087)
apache/ignite: DEVNOTES.txt

Apache Ignite Maven Build Instructions
======================================
1) Compile and install:

        ./mvnw clean install -Pall-java,all-scala,licenses -DskipTests

>    or if you have to built Apache Ignite with Apache Ignite .NET use following command:

Reviewer: ```suggestion
   or if you have to build .NET part as well, use the following command:
```

   (Note that 'doxygen', '.NET Core SDK 3.1', 'PowerShell' should be installed before running this command.)

>         ./mvnw clean install -Pall-java,all-scala,licenses,platforms -DskipTests -DclientDocs

Reviewer: I've tried `mvn clean package  -Pplatforms -DskipTests` and got `[WARNING] The requested profile "platforms" could not be activated because it does not exist.` - am I doing something wrong?

Author: The message you mentioned below appears due to the `ignite-dotnet` module itself doesn't contain any information about `platforms` profile, however we still need to add `ignite-dotnet` and `ignite-cpp` modules to the maven reactor.

If you're using the command below to build everything this message won't appear:
`mvn clean package -Pplatforms -DskipTests`

If you're using the command to build particular module, you'll see the message due to this module doesn't have any info about the `platforms` profile:
`mvn clean package -am -pl modules/platforms/dotnet -Pplatforms -DskipTests`

Apache Ignite with LGPL Maven Build Instructions
================================================
1) Compile and install:

        ./mvnw clean install -Pall-java,all-scala,licenses -DskipTests

>    or if you have to built Apache Ignite with Apache Ignite .NET use following command:

Reviewer: ```suggestion
   or if you have to build .NET part as well, use the following command:
```
